src/interpreter.rs

```diff
@@ -4,12 +4,12 @@ use std::collections::HashMap;
 type Stack<T> = Vec<T>;
 
 #[derive(Default)]
-pub struct Interpreter<'a, T> {
+pub struct Interpreter<'a, 'b, T: 'b> {
```

Comment:
The lifetime of the Word may be different from the lifetime of the key to look it up